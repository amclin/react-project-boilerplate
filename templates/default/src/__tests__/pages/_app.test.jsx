/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import { render } from '@testing-library/react'
import App from '../../pages/_app'

jest.mock('../../utils/apollo', () => ({
  withApollo: Component => ({ ...pageProps }) => (
    <div data-testid="apollo-wrapper">
      <Component {...pageProps} />
    </div>
  )
}))

const mockComponent = () => <div data-testid="mock-component">Ok!</div>

describe('App', () => {
  it('controls the root NextJS app', () => {
    const { asFragment, getByTestId } = render(
      <App Component={mockComponent} pageProps={{}} />
    )

    expect(getByTestId('apollo-wrapper')).toBeTruthy()
    expect(getByTestId('mock-component')).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
})
