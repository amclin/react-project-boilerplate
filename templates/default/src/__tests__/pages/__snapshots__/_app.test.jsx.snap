// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App controls the root NextJS app 1`] = `
<DocumentFragment>
  <div
    data-testid="apollo-wrapper"
  >
    <div
      data-testid="mock-component"
    >
      Ok!
    </div>
  </div>
</DocumentFragment>
`;
